feat: It is possible to pass attributes to embedded iframe

ra002890 opened the pull request:
feat: It is possible to pass attributes to embedded iframe in embedded-sdk

### SUMMARY
It must be possible to control the embedded iframe inserted by the embedded-sdk. With this change we can pass any html attribute to it, including id.

### TESTING INSTRUCTIONS

1. Pass a map containing keys(html attribute names) and values(html attribute values) to the parameter iframeAttributes of EmbedDashboardParams.
2. Check on the rendered page if the iframe got passed attributes.

```javascript
 embedDashboard({
            id: "8", // given by the Superset embedding UI
            supersetDomain: "http://127.0.0.1:9000",
            mountPoint: document.getElementById("my-superset-container")!, // any html element that can contain an iframe
            iframeAttributes: new Map<string, string>([["id", "my-test-frame"], ["className", "my-test-frame-class"]]),
            fetchGuestToken: () => fetchGuestTokenFromBackend(),
        });
```
### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

ra002890 committed:
feat: It is possible to pass attributes to embedded iframe

ra002890 committed:
feat: It is possible to pass attributes to embedded iframe

suddjian commented on superset-embedded-sdk/src/index.ts:
```diff
@@ -36,6 +36,8 @@ export type EmbedDashboardParams = {
   /** The html element within which to mount the iframe */
   mountPoint: HTMLElement
+  /** An id to identify the create iframe that will embed Superset */
+  iframeAttributes: Map<string, string> 
```
This will require instantiating a `Map()` object. We should probably allow passing key-value pairs such as `{ class: "some-class" }`. Let's change the type to use `Record` which represents this. (This will also require changing the `forEach` construction below)

```suggestion
  iframeAttributes: Record<string, string> 
```

ra002890 replied:
From the first time I was trying to force only Iframe related attributes to be passed here, and I was not able to find a good way to do that, but with your suggestion I have found the Partial type and I could implement it. What do you think about my new implementation where only valid iframe attributes can be passed to it?

suddjian replied:
Oh, cool idea

ra002890 committed:
Merge branch 'master' into feature/pass_attributes_to_iframe

ra002890 committed:
Just attributes related with iframes can be added

ra002890 committed:
Just attributes related with iframes can be added

geido committed:
fix(Explore): Fix cache timeout field not being saved and unit tests (#18738)

* fix cache timeout

* Add unit tests

wiktor2200 committed:
fix: bump Helm chart release version (#18751)

Commit:
Adding OAF countries (#18745)

geido committed:
fix(Dashboard): Fix blinking native filters on zoom (#18717)

* Fix blinking filters

* Improve css pointer

rusackas committed:
chore: bumping caniuse-lite from 1.0.30001257 to 1.0.30001312 (#18705)

pkdotson committed:
chore: move data nav menu to plus menu (#18629)

* more data nav menu

* fix lint and fix nav css

* update test and remove icons

* Update superset-frontend/src/views/components/Menu.test.tsx

Co-authored-by: Hugh A. Miles II <hughmil3s@gmail.com>

* Apply suggestions from code review

* use backend app.link to show new nav changes

* fix lint

* update test

* usetheme and remove chaining

* add more suggestions

* fix lint

* add allowed extensions to bootstrap and hard code links

* remove backend links

* fix test

* add extensions to frontend conf

* fix test and add be changes

* test is python test passes

* update python test and reremove app links

* fix ts and add t's

Co-authored-by: Hugh A. Miles II <hughmil3s@gmail.com>

dependabot[bot] committed:
chore(deps): bump url-parse from 1.4.7 to 1.5.7 in /superset-frontend (#18771)

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.4.7 to 1.5.7.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.4.7...1.5.7)

---
updated-dependencies:
- dependency-name: url-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump dns-packet from 1.3.1 to 1.3.4 in /superset-frontend (#18770)

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

---
updated-dependencies:
- dependency-name: dns-packet
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump tmpl from 1.0.4 to 1.0.5 in /superset-frontend (#18768)

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump node-fetch in /superset-frontend (#18766)

Bumps [node-fetch](https://github.com/node-fetch/node-fetch) from 2.6.1 to 2.6.7.
- [Release notes](https://github.com/node-fetch/node-fetch/releases)
- [Commits](https://github.com/node-fetch/node-fetch/compare/v2.6.1...v2.6.7)

---
updated-dependencies:
- dependency-name: node-fetch
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump nanoid from 3.1.30 to 3.3.1 in /docs (#18763)

Bumps [nanoid](https://github.com/ai/nanoid) from 3.1.30 to 3.3.1.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.1.30...3.3.1)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.14.5 to 1.14.8 in /docs (#18760)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.14.5 to 1.14.8.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.14.5...v1.14.8)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump shelljs from 0.8.4 to 0.8.5 in /docs (#18759)

Bumps [shelljs](https://github.com/shelljs/shelljs) from 0.8.4 to 0.8.5.
- [Release notes](https://github.com/shelljs/shelljs/releases)
- [Changelog](https://github.com/shelljs/shelljs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/shelljs/shelljs/compare/v0.8.4...v0.8.5)

---
updated-dependencies:
- dependency-name: shelljs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump path-parse from 1.0.6 to 1.0.7 in /superset-frontend (#18772)

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.6 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump underscore from 1.12.0 to 1.12.1 in /superset-frontend (#18762)

Bumps [underscore](https://github.com/jashkenas/underscore) from 1.12.0 to 1.12.1.
- [Release notes](https://github.com/jashkenas/underscore/releases)
- [Commits](https://github.com/jashkenas/underscore/compare/1.12.0...1.12.1)

---
updated-dependencies:
- dependency-name: underscore
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump follow-redirects in /superset-frontend (#18764)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.14.3 to 1.14.8.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.14.3...v1.14.8)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump shelljs from 0.8.4 to 0.8.5 in /superset-frontend (#18767)

Bumps [shelljs](https://github.com/shelljs/shelljs) from 0.8.4 to 0.8.5.
- [Release notes](https://github.com/shelljs/shelljs/releases)
- [Changelog](https://github.com/shelljs/shelljs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/shelljs/shelljs/compare/v0.8.4...v0.8.5)

---
updated-dependencies:
- dependency-name: shelljs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

zhaoyongjie committed:
refactor: postprocessing move to unit test (#18779)

ra002890 committed:
Removed extra space and added pre-commit

ra002890 committed:
Removed extra space and added pre-commit

dependabot[bot] committed:
chore(deps): bump ws from 5.2.2 to 5.2.3 in /superset-frontend (#18769)

Bumps [ws](https://github.com/websockets/ws) from 5.2.2 to 5.2.3.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/5.2.2...5.2.3)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump y18n from 4.0.0 to 4.0.3 in /superset-frontend (#18765)

Bumps [y18n](https://github.com/yargs/y18n) from 4.0.0 to 4.0.3.
- [Release notes](https://github.com/yargs/y18n/releases)
- [Changelog](https://github.com/yargs/y18n/blob/y18n-v4.0.3/CHANGELOG.md)
- [Commits](https://github.com/yargs/y18n/compare/v4.0.0...y18n-v4.0.3)

---
updated-dependencies:
- dependency-name: y18n
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

suddjian reviewed (commented):
We'll also need to increase the version in package.json so that this gets built and released when merged. For now I've just been increasing the `-alpha.x` component of the version.

dependabot[bot] committed:
chore(deps): bump cached-path-relative (#18761)

Bumps [cached-path-relative](https://github.com/ashaffer/cached-path-relative) from 1.0.2 to 1.1.0.
- [Release notes](https://github.com/ashaffer/cached-path-relative/releases)
- [Commits](https://github.com/ashaffer/cached-path-relative/commits)

---
updated-dependencies:
- dependency-name: cached-path-relative
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prosdev0107 committed:
fix(delete-modal): Changed the color of the help text with grayScale.dark1 in DeleteModal (#18786)

* fix(delete-modal): Changed the color of the help text with grayScale.dark1 in DeleteModal

* Update superset-frontend/src/components/DeleteModal/index.tsx

Co-authored-by: Evan Rusackas <evan@preset.io>

hughhhh committed:
fix: update standalone url for slack reports (#18756)

* update url for slack reports

* fix test

* update text to be dynamic

EugeneTorap committed:
refactor: migrate ExploreCtasResultsButton component to typescript (#18142)

* Move component to FC & tsx

* Refactoring

* Refactoring

* Refactoring

* Refactoring

* Refactoring

* Refactoring

* Refactoring

* Fix type issue

* Fix type issue

* Fix ResultSet type issue

* Refactoring RootState

rusackas committed:
chore(FF): adding generic X axis Feature Flag (default: False) (#18801)

dependabot[bot] committed:
chore(deps): bump url-parse from 1.5.3 to 1.5.7 in /docs (#18804)

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.5.3 to 1.5.7.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.5.3...1.5.7)

---
updated-dependencies:
- dependency-name: url-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump hosted-git-info in /superset-frontend (#18805)

Bumps [hosted-git-info](https://github.com/npm/hosted-git-info) from 2.7.1 to 2.8.9.
- [Release notes](https://github.com/npm/hosted-git-info/releases)
- [Changelog](https://github.com/npm/hosted-git-info/blob/v2.8.9/CHANGELOG.md)
- [Commits](https://github.com/npm/hosted-git-info/compare/v2.7.1...v2.8.9)

---
updated-dependencies:
- dependency-name: hosted-git-info
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

xiangfu0 committed:
Update pinot docs with fixed pinot connection (#18800)

xiangfu0 committed:
Adding pinot logo to ReadMe (#18798)

zhaoyongjie committed:
refactor: move date_parser to unittest (#18810)

zhaoyongjie committed:
fix: contribution operator meets nan value (#18782)

jacobcroope committed:
docs: Added reference and link to configure instructions for docker-compose… (#18815)

Co-authored-by: Jacob Roope <[jacobcroope@gmail.com]>

prosdev0107 committed:
fix(sqllab): Removed the tooltip from CopyToClipboard button in sqllab (#18749)

* fix(15849): Removed the tooltip from CopyToClipboard button in sqllab

* chore(sqllab): added props for tooltip in CopyToClipboard component

* fix(sqllab): Added arg to storybook and refactor the component

* fix(sqllab): added a test case for hideTooltip

srinify committed:
docs: add_db_logos (#18818)

* docs: added yugabytedb and teradata logos in all the right places

* docs: added yugabytedb and teradata logos in all the right places

ra002890 committed:
Update embedded sdk version

ra002890 committed:
Update embedded sdk version

ra002890 commented:
> We'll also need to increase the version in package.json so that this gets built and released when merged. For now I've just been increasing the `-alpha.x` component of the version.

I think it is done!

srinisubramanian committed:
Update INTHEWILD.md (#18824)

Added Unvired usage in enterprise tech

prosdev0107 committed:
fix(altered-modal): displayed the metric value in altered modal correctly (#18813)

Co-authored-by: Evan Rusackas <evan@preset.io>

Commit:
Update configuring-superset.mdx (#18819)

Fixes typo dateime -> datetime

sujiplr committed:
fix(mssql): support top syntax for limiting queries (#18746)

* SQL-TOP Fix For Database Engines

MSSQL is not supporting LIMIT syntax in SQLs. For limiting the rows, MSSQL having a different keyword TOP. Added fixes for handling the TOP and LIMIT clauses based on the database engines.

* Teradata code for top clause handling removed from teradata.py

Teradata code for top clause handling removed from teradata.py file, since we added generic section in base engine for the same.

* Changes to handle CTE along with TOP in complex SQLs

Added changes to handle TOP command in CTEs, for DB Engines which are not supporting inline CTEs.

* Test cases for TOP unit testing in MSSQL

Added multiple unit test cases for MSSQL top command handling and also along with CTEs

* Corrected the select_keywords name key in basengine

Corrected the select_keywords name key in basengine

* Changes based on as per review.

made the required corrections based on code review to keep good code readability and code cleanliness.

* Review changes to correct lint and typo issues

Made the changes according to the review comments.

* fix linting errors

* fix teradata tests

* add coverage

* lint

* Code cleanliness

Moved the top/limit flag check from sql_lab to core.

* Changed for code cleanliness

Changes for keeping code cleanliness

* Corrected lint issue

Corrected lint issue.

* Code cleanliness

Code cleanliness

Co-authored-by: Ville Brofeldt <ville.v.brofeldt@gmail.com>

stephenLYZ committed:
chore: unify usages of nvd3 library (#18828)

ra002890 commented:
> We'll also need to increase the version in package.json so that this gets built and released when merged. For now I've just been increasing the `-alpha.x` component of the version.

I think it is done!

geido committed:
Fix annotation cache (#18788)

saurabh3091 committed:
stop query on the processing engine when user clicks STOP on SQL Lab (#18789)

Co-authored-by: Saurabh Gulati <saurabh.gulati@fedex.com>

rusackas committed:
chore: npm audit fix in cypress-base (#18820)

geido committed:
fix: Position of arrows in Table chart (#18739)

* Fix arrow position

* Improve position

* Add right position

* Improve styling

zhaoyongjie committed:
fix(key-value): lost url_params after long-url feature (#18846)

kgabryje committed:
feat(native-filters): Re-arrange controls in FilterBar (#18784)

* feat(native-filters): Re-arrange controls in FilterBar

* Typo fix

* Add tests

* Fix tests

* Address PR reviews

* Add 1px bottom padding to icon

* Fix test

* Tiny refactor

* Change buttons background

* Add hover state to Clear all button

* Fix Clear All button logic so it clears all selections

* Remove redundant useEffect

* Set zindex higher than loading icon

* Fix scrolling issue

villebro committed:
docs: add GA conf to config (#18861)

* docs: add GA conf to config

* add license and bump docusaurus

kgabryje committed:
feat(native-filters): Don't scroll main window when scrolling filter bar (#18876)

michaelzusev committed:
docs: fix toc position scroll bug (#18868) (#18878)

Commit:
fix: FilterBox date endpoints to use [inclusive, exclusive) (#18857)

kgabryje committed:
feat(native-filters): Implement filter cards (#18874)

* feat(native-filters): Filter cards first pass

* Add dependencies row and lots of improvements

* Display tooltip correctly

* Styling fix

* Remove accidentaly added changes

* Address comments

* Fix type

* Fixes

geido committed:
Fix database list (#18886)

corbinrobb committed:
chore(tests-backend): migrate snowflake and sqlite to unit tests (#18854)

* migrate snowflake and sqlite

* change structure to better match the other unit tests

Co-authored-by: Corbin Robb <corbin@Corbins-MacBook-Pro.local>

michael-hoffman-26 committed:
docs(Import API): update function description, overwrite real meaning (#18882)

Co-authored-by: michael_hoffman <michael.hoffman@nielsen.com>

ktmud committed:
fix(dashboard): incorrect chart error with slow dataset api request (#18852)

* fix(dashboard): incorrect chart error with slow datasource api request

* update comments

* Add key to styled loading component

vuongxuongminh committed:
feat: allow adding pod labels (#18848)

paulocauca committed:
fix: Source from Apache Foundation's website link (#18894)

This PR fixes a link to latest version on Download the source from Apache Foundation's website link

from : https://dist.apache.org/repos/dist/release/superset/1.0.0/
to: https://dist.apache.org/repos/dist/release/superset/1.4.1/

dependabot[bot] committed:
chore(deps-dev): bump css-loader in /superset-frontend (#17977)

Bumps [css-loader](https://github.com/webpack-contrib/css-loader) from 6.2.0 to 6.5.1.
- [Release notes](https://github.com/webpack-contrib/css-loader/releases)
- [Changelog](https://github.com/webpack-contrib/css-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/css-loader/compare/v6.2.0...v6.5.1)

---
updated-dependencies:
- dependency-name: css-loader
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

villebro committed:
docs: remove old viz plugin docs (#18914)

villebro committed:
fix(docs): add missing static files to build (#18920)

villebro committed:
refactor(docs): clean up deploy of static files (#18923)

hughhhh committed:
add force condition for exporting csv (#18911)

suddjian committed:
feat: On window focus, redirect to login if the user has been logged out (#18773)

* /me api

* test it

* watch for window activation and check auth

* simplify

* more comment

* making ci happy

* mypy should ignore tests

jinghua-qa committed:
test(native filter): add new native filter test for time column filter (#18850)

* add new native filter test for time column filter

codemaster08240328 committed:
fix: filter indicator pointing issue on popover (#18853)

michael-hoffman-26 committed:
refactor(ImportModelsCommand): clean code, extract logic into methods (#18866)

Co-authored-by: michael_hoffman <michael.hoffman@nielsen.com>

betodealmeida committed:
feat: new dataset/table/column models (#17543)

* feat: new models for SIP-68

* feat: new dataset models DB migration

* Add shadow write (WIP)

* Physical dataset shadow write (WIP)

* Virtual dataset shadow write (WIP)

* Update migration to populate models

* Cascade delete columns

* after_delete

* Update hook

* Add metric test

* Do not rename tables

* Small fixes

* Fix migration

* Fix tests

* Fix more tests

* Even more tests

* Addressing comments (WIP)

* Fix migration

* Rebase and update head

* Fix unit tests

* Add external management cols

* Small fixes

betodealmeida committed:
chore: update UPDATING re: SIP-68 (#18935)

hughhhh committed:
feat: DBC-UI Globally available across the app 🌎 (#18722)

* more data nav menu

* fix lint and fix nav css

* update test and remove icons

* Update superset-frontend/src/views/components/Menu.test.tsx

Co-authored-by: Hugh A. Miles II <hughmil3s@gmail.com>

* Apply suggestions from code review

* use backend app.link to show new nav changes

* fix lint

* update test

* usetheme and remove chaining

* add more suggestions

* fix lint

* working global db connection

* add allowed extensions to bootstrap and hard code links

* remove backend links

* fix test

* apply stashed gsheets

* fix check for google sheets

* setup gsheets

* add extensions to frontend conf

* fix test and add be changes

* remove package json changes

* test is python test passes

* update python test and reremove app links

* fix tslint issues

* fix other linting tools

* fix pylint

* fix test

* fix

* refactor

* fix lint

* working fixed test

* clean up test

* address concerns

* address concerns

* change to tenarary

Co-authored-by: Phillip Kelley-Dotson <pkelleydotson@yahoo.com>

Commit:
fix(sqllab): Un-render deselected tabs (#18934)

pkdotson committed:
chore: add intersect method from backend to fontend for upload extensions (#18811)

* chore: add intersect method from be to font end

* fix lint

* add suggestion

* fix python test

* run precommit

* fix pytlint

* update changes from masters

codemaster08240328 committed:
fix Metadata browser in SQL not render nicely in Safari (#18855)

codemaster08240328 committed:
fix: Monospacing errors in dashboards & charts (#18796)

* fix: Monospacing errors in dashboards & charts

* removed unnecessary styling

jinghua-qa committed:
test(Native Filter): User can create parent filters using "Filter is hierarchical (#18915)

* add new native test for parent filter

stephenLYZ committed:
fix(native-filter): fix required filters of a dashboard won't load properly (#18695)

stephenLYZ committed:
fix(deck.gl): multiple layers map size is shrunk (#18939)

michael-s-molina committed:
chore: Moves components out of the commons folder (#18931)

* refactor: Moves components out of the commons folder

* Fixes tests

codemaster08240328 committed:
fix: null schema issue in saved queries (#18849)

john-bodley committed:
fix: bump Helm chart release version (#18751) (#18758)

Co-authored-by: wiktor2200 <wiktor2200@users.noreply.github.com>

AAfghahi committed:
fix(sql Lab tabs): Empty SQL Lab tabs (#18817)

* Empty SQL table message on zero tabs

* sql editor no editor tab bug fix

* Revert Error message

* empty state tab state

* added a unit test

* addressed reviews

* kasia feedback

Co-authored-by: Yahya Kayani <yahyakiani1@gmail.com>

villebro committed:
fix(dataset): handle missing database in migration (#18948)

ra002890 committed:
Merge branch 'feature/pass_attributes_to_iframe' of github.com:ra002890/superset into feature/pass_attributes_to_iframe

ra002890 commented:
It took too long to be accepted and I made a mistake in rebase it in my fork...I will close it and open a new one!
